tests/virt_autotest/prepare_alp_host.pm

```diff
@@ -0,0 +1,50 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: This file does the necessary setup and checks for alp host to do virtualization test.
+# Maintainer: alice <xlai@suse.com>, qe-virt@suse.de
+
+package prepare_alp_host;
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+
+    # Validate that kernel-default is installed and kvm module is loaded.
+    # Note: the alp host unattended installation will include kernel-default
+    #       package installation. See poo#118939 for more details.
+    assert_script_run(q@rpm -qa | grep kernel-default-[0-9]@);
+    save_screenshot;
+    assert_script_run(q@lsmod |grep kvm@);
+    save_screenshot;
+    record_info("KVM module is successfully loaded.");
+
+    # Download alp vm ignition config files
+    assert_script_run("mkdir -p /var/lib/libvirt/images");
+    my $cmd = "curl -L "
+      . data_url("virt_autotest/guest_unattended_installation_files/VIRT_TEST_VM_config.ign")
+      . " -o /var/lib/libvirt/images/VIRT_TEST_VM_config.ign";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    assert_script_run("ls -latr /var/lib/libvirt/images/");
```

Comment:
You should list desire file out instead of directory I think

Reply:
Same effect I think. If script_retry passes, it already actually ensures the target file is there. ALP test currently starts from existing host, which can have existing data in this dir. So listing it can give additional info here .

Reply:
Listing out the folder can really give info about, but it does not guarantee what you want because it does not assert on it at all. You may think creating directory succeeded, downloading succeeded, then the file must exist, but what you do here does not match your thinking IMHO. If you only need show info, use record_info, if you want to make sure the file is there, then list that file out.  Listing out the folder only does not look like anything, it just assert the folder existence.

Reply:
Is it really worth the discussion effort on this minor thing?
Here the `assert_script_run("ls -latr /var/lib/libvirt/images/")` is to show the files under this dir. I care about the assert and the saved screenshot for it. Yes, record_info can achieve this, but this line does it too I think. To do it with record_info, one has to call script_output in addition to record_info, so simpler just use assert_script_run.

Reply:
The default network on alp can be seen in http://10.67.129.185/tests/537#step/setup_kvm_container/147. So you can see, on baremetal alp host, no default bridge br0 at all. Within libvirtd container, the "default" network is sle_network, a nat based network which has nothing to do with br0 either. Even if we create br0 during host preparation but before libvirtd container is created, the br0 is not hurt at all. 
This function cleans VIRTUAL networks, rather than physical networks. So the physical networks on host are still there.
Removing the "default" virtual network won't hurt the libvirtd container connectivity either, and it will exclude any potential impact to some of our hard-coded existing code from sles. So that's why in previous PRs, this solution was taken. This logic is not introduced in this PR.
I am open to change the solution, as long as there are good reasons. But not convinced by above comments. It will need a separate ticket to handle all possible work for it to not break any existing alp test.

Reply:
Well, is it worth the discussion effort on this minor thing ;-) ?

The purpose of the line is to get the image dir contents. Assert should work for this ls command, although the main purpose is not to assert, but to see the saved screenshot from assert_script_run. Yes, record_info can do it too, but it needs to also call script_output. So this way is simpler I think.

I would keep it if you do not insist on change.

Reply:
I will not do things this way. But if you insist, you can keep it this way.

Reply:
Ok, thanks. I will keep it then. But I got your point :).